File: my-folder/0215-kth-largest-element-in-an-array/solution.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        max_heap = [-nums[i] for i in range(len(nums))]
        heapq.heapify(max_heap)
        k_largest = k
        k_largest_value = 0
        while (k_largest > 0):
            k_largest_value = heapq.heappop(max_heap)
            k_largest -= 1
        return -1 * k_largest_value

        

File: my-folder/0215-kth-largest-element-in-an-array/solution.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        if(nums.length == 0){
            return 0;
        }
        PriorityQueue<Integer> elements = new PriorityQueue<>();
        for(int num: nums){
            elements.add(num);
            while(elements.size() > k){
                elements.poll();
            }
        }
        
        return elements.peek();
    }
}

// appproach is to use heaps 
// create a minheap with -values then pop the elements k times and return the kth element
